kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -259,6 +259,7 @@ class ResidualBasedNewtonRaphsonStrategy
      
     virtual ~ResidualBasedNewtonRaphsonStrategy()
     {
+        Clear();
```

Comment:
Hey Jordi, a problem occurs with the MultiLevelSolver when the order of destruction is not carefully controlled. The member function Solve() creates a multilevel preconditioner of the matrix in mpA. This preconditioner stores a reference to the matrix which it tries to access when it is destructed. We can't prevent the matrix in mpA from destructing early by storing a smart pointer in the linear solver because the matrix is passed as a reference to Solve(). Therefore, we need to call the Clear() function of the linear solver before the matrix is freed.